src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_transformers.py

```diff
@@ -102,7 +102,17 @@ def _transformer(result):
 
 
 def transform_storage_list_output(result):
-    return list(result)
+    if isinstance(result, dict):
+        next_marker = result['next_marker']
+        result = result['generator']
+    else:
+        next_marker = result.next_marker
+
+    def set_marker(item):
+        item.next_marker = next_marker
+        return item
+
+    return list(map(set_marker, result))
```

Comment:
Use list comprehensions instead of map
https://www.safaribooksonline.com/library/view/python-cookbook/0596001673/ch01s11.html